openmc/universe.py

```diff
@@ -823,12 +825,16 @@ def __init__(self,
                  universe_id=None,
                  name='',
                  auto_geom_ids=False,
-                 auto_mat_ids=False):
+                 auto_mat_ids=False, 
+                 mat_assignment={}):
         super().__init__(universe_id, name)
         # Initialize class attributes
         self.filename = filename
         self.auto_geom_ids = auto_geom_ids
         self.auto_mat_ids = auto_mat_ids
+        self.mat_assignment = mat_assignment
+        self._num_instances = 0
```

Comment:
Thinking about this more I think we should place this attribute on the `DAGMCPseudoCell`/`DAGMCCell` class. It may be a little redundant, but it will reduce the amount of DAGMC-specific code at higher levels of the `Model` and `Geometry` classes.